app/uk/gov/hmrc/economiccrimelevyregistration/services/RegistrationValidationService.scala

```diff
@@ -326,4 +337,59 @@ class RegistrationValidationService @Inject() (clock: Clock, schemaValidator: Sc
 
   private def missingErrorMessage(missingDataDescription: String): String = s"$missingDataDescription is missing"
 
+  private def validateOtherEntity(
+    registration: Registration
+  ): ValidationResult[Either[EclSubscription, Registration]] = {
+    val otherEntityJourneyData = registration.otherEntityJourneyData
+    (
+      validateAmlSupervisor(registration),
+      validateOptExists(registration.businessSector, "Business sector"),
+      validateContactDetails("First", registration.contacts.firstContactDetails),
+      validateSecondContactDetails(registration.contacts),
+      validateAmlRegulatedActivity(registration),
+      validateOptExists(registration.relevantAp12Months, "Relevant AP 12 months choice"),
+      validateOptExists(registration.relevantApRevenue, "Relevant AP revenue"),
+      validateConditionalOptExists(
+        registration.relevantApLength,
+        registration.relevantAp12Months.contains(false),
+        "Relevant AP length"
+      ),
+      validateRevenueMeetsThreshold(registration),
+      validateOptExists(registration.optOtherEntityJourneyData, "other entity data"),
+      validateCharity(registration, otherEntityJourneyData)
+    ).mapN {
+      (
+        _,
+        _,
+        _,
+        _,
+        _,
+        _,
+        _,
+        _,
+        _,
+        _,
+        _
+      ) => Right(registration)
+    }
+  }
+
+  private def validateCharity(
```

Comment:
```suggestion
  private def validateOtherEntityJourneyData(
```

I'm thinking it makes more sense to call this method this - because it will be updated going forward to contain the validation for the other types as well

Reply:
Refactored code.